tools/resourcedocsgen/cmd/metadata.go

```diff
@@ -47,264 +45,315 @@ var featuredPackages = []string{
 			}
+		}
 
-			return nil
-		},
+		return stderrors.Join(errs...)
 	}
 
+	return cmd
+}
+
+func packageMetadataFromGitHubCmd(metadataDir, packageDocsDir *string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "from-github <args>",
+		Short:        "Generate package metadata from Pulumi schema",
+		SilenceUsage: true,
+	}
+
+	// Inputs (required)
+
+	var repoSlug repoSlug
+	var version string
 	cmd.Flags().Var(&repoSlug, "repoSlug", "The repository slug e.g. pulumi/pulumi-provider")
+	cmd.Flags().StringVar(&version, "version", "", "The version of the package")
+
+	contract.AssertNoErrorf(stderrors.Join(
+		cmd.MarkFlagRequired("repoSlug"),
+		cmd.MarkFlagRequired("version"),
+	), "impossible")
+
+	// Inputs (optional)
+
+	var providerName string
+	var schemaFile string
+	var publisher string
+
 	cmd.Flags().StringVar(&providerName, "providerName", "", "The name of the provider e.g. aws, aws-native. "+
 		"Required when there is no schemaFile flag specified.")
-	cmd.Flags().StringVar(&rawSource.schemaFileURL, "schemaFileURL", "",
-		`The URL from which the schema can be retrieved.
-
-schemaFileURL takes precedence over schemaFile.`)
-	cmd.Flags().StringVar(&rawSource.indexFileURL, "indexFileURL", "",
-		`The URL from which the docs/_index.md file can be retrieved.`)
-	cmd.Flags().StringVar(&rawSource.installationConfigurationFileURL, "installationConfigurationFileURL", "",
-		`The URL from which the docs/installation-configuration.md file can be retrieved.`)
 	cmd.Flags().StringVarP(&schemaFile, "schemaFile", "s", "",
 		"Relative path to the schema.json file from the root of the repository. If no schemaFile is specified,"+
 			" then providerName is required so the schemaFile path can "+
 			"be inferred to be provider/cmd/pulumi-resource-<providerName>/schema.json")
-	cmd.Flags().StringVar(&version, "version", "", "The version of the package")
-	cmd.Flags().StringVar(&categoryStr, "category", "", fmt.Sprintf("The category for the package. Value must "+
-		"match one of the keys in the map: %v", pkg.CategoryNameMap))
 	cmd.Flags().StringVar(&publisher, "publisher", "", "The publisher's display name to be shown in the package. "+
 		"This will default to Pulumi")
-	cmd.Flags().StringVar(&title, "title", "",
-		"The display name of the package. If omitted, the name of the package will be used")
-	cmd.Flags().BoolVar(&component, "component", false,
-		"Whether or not this package is a component and not a provider")
-	cmd.Flags().StringVar(&metadataDir, "metadataDir", "themes/default/data/registry/packages",
-		"The location to save the metadata - this will default to the folder "+
-			"structure that the registry expects (themes/default/data/registry/packages)")
-	cmd.Flags().StringVar(&packageDocsDir, "packageDocsDir", "",
-		"The location to save the package docs - this will default to the folder "+
-			"structure that the registry expects (themes/default/content/registry/packages)")
+
+	// Apply complex defaults
```

Comment:
What is complex about it?

Reply:
Complex was the wrong word. I have changed it to:

```go
// Apply default values that depend on other inputs.
```